api/v1/server/handlers/v1/workflow-runs/trigger.go

```diff
@@ -122,6 +122,8 @@ func (t *V1WorkflowRunsService) V1WorkflowRunCreate(ctx echo.Context, request ge
 		return nil, err
 	}
 
+	ctx.Set("correlationId", sqlchelpers.UUIDToStr(rawWorkflowRun.WorkflowRun.ExternalID))
```

Comment:
I'd be strongly in favor of searching for a specific key in the additional metadata as well, and setting that as the correlation ID if it exists. Note that I think correlation ID and resource ID are distinct and we should probably store both